### ISSUE TITLE
Have scheme work for collections of items

### ISSUE THREAD

**Issue by tommoor** [deloreanjs/delorean#75]
It would be nice if the scheme validation and calculation methods also worked for a collection of items as this is typically how Flux stores are arranged. If we discuss some syntax I can probably take this on... my thinking is another attribute on the stores isCollection: true for example.

**Comment by darcyadams** [comment]
agreed. This bugs me as well. I find myself operating directly on `scheme` properties, usually when dealing with arrays.

Not sure I follow exactly what you are proposing. Other frameworks, such as Backbone, have solved this by mixing in a library like underscore, and I'm not sure I want to introduce a dep like that. Can you be more specific about how you envision this working?

**Comment by tommoor** [comment]
From an API point of view, i might suggest something like this:

Store is a single 'model':
```
var Artist = Flux.createStore({
  scheme: {
    firstName: {
      default: 'Unknown'
    },
    lastName: {
      default: 'Artist'
    }
  }
});
```

Store is a collection of 'models':
```
var Artists = Flux.createStore({
  collection: true,
  scheme: {
    firstName: {
      default: 'Unknown'
    },
    lastName: {
      default: 'Artist'
    }
  }
});
```

In my own stores of this nature I also have the methods add, remove and update that could work something like:

```
Artists.add({
  id: 1,
  firstName: 'Elvis',
  lastName: 'Presley'
});

Artists.update({id: 1}, {
  firstName: 'Lisa'
});

Artists.remove({id: 1});
```

Calling set on this store would throw an error. Maybe this is overkill?
